Support ircs as an protocol

KaiserBarbarossa opened the pull request:
Already implemented in Inyoka but not yet deployed.

Additionally:

Why is http/https and ftp/ftps seperate here?

https://github.com/inyokaproject/inyokaedit/blob/9dff8f0d12d05b1af82a22ef17dc707353464e4e/application/parser/parselinks.cpp#L111-L112

Couldn’t that be unified? If so, I’d add this to that PR.

KaiserBarbarossa committed:
Support ircs as an protocol

ElTh0r0 commented:
Many thanks!

> Why is http/https and ftp/ftps seperate here?
> https://github.com/inyokaproject/inyokaedit/blob/9dff8f0d12d05b1af82a22ef17dc707353464e4e/application/parser/parselinks.cpp#L111-L112
> Couldn’t that be unified? If so, I’d add this to that PR.

Yes, I think this can be unified! I assume this was caused by my very basic/limited regular expression knowledge :)

KaiserBarbarossa committed:
Unify regex

Changes the regex

        QString::fromLatin1("\\[{1,1}\\b(http|https|ftp|ftps|file|ssh|mms|svn"
                            "|git|dict|nntp|ircs?|rsync|smb|apt)\\b://"));

to

        QString::fromLatin1("\\[{1,1}\\b(https?|ftps?|file|ssh|mms|svn"
                            "|git|dict|nntp|ircs?|rsync|smb|apt)\\b://"));

KaiserBarbarossa commented:
> Yes, I think this can be unified! I assume this was caused by my very basic/limited regular expression knowledge :)

Done :)
